src/docs.mts

```diff
@@ -1,84 +1,183 @@
   });
 
-  const transformed = `# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT. Please edit corresponding file in src/docs.\n${remark.stringify(
-    out
-  )}`;
+  // Add the AUTOGENERATED_TEXT to the start of the file
+  const transformed = `${AUTOGENERATED_TEXT}\n${remark.stringify(out)}`;
+
   verifyAndCopy(
     file,
     prettier.format(transformed, {
-      parser: 'markdown',
+      parser: "markdown",
       useTabs: false,
       tabWidth: 2,
-      endOfLine: 'auto',
+      endOfLine: "auto",
       printWidth: 100,
-      singleQuote: true,
+      singleQuote: true
     })
   );
 };
 
+/**
+ * Transform a HTML file and write the transformed file to the directory for published documentation
+ *   - add the text that says the file is automatically generated
+ *   Verify that the file has been changed and write out the changes
+ *
+ * @param filename {string} name of the HTML file to transform
+ */
+const transformHtml = (filename: string) => {
+
+  /**
+   * Insert the '...auto generated...' comment into an HTML file after the <html> element
+   *
+   * @param fname {string} file name that should have the comment inserted
+   * @returns {string} the contents of the file with the comment inserted
+   */
+  function insertAutoGeneratedComment(fname: string): string {
+    const fileContents = readSyncedUTF8file(fname);
+    const jsdom = new JSDOM(fileContents);
+    const htmlDoc = jsdom.window.document;
+    const autoGeneratedComment = jsdom.window.document.createComment(AUTOGENERATED_TEXT);
+
+    let rootElement = htmlDoc.documentElement;
+    rootElement.prepend(autoGeneratedComment);
+    return jsdom.serialize();
+  }
+
+  let transformedHTML = insertAutoGeneratedComment(filename);
+  verifyAndCopy(filename, transformedHTML);
+};
+
 (async () => {
-  const mdFiles = await globby(['./src/docs/**/*.md'], { dot: true });
-  mdFiles.forEach(transform);
-  const nonMDFiles = await globby(['src/docs/**', '!**/*.md'], { dot: true });
-  nonMDFiles.forEach((file) => {
+  console.log("Transforming markdown files...");
+  const mdFiles = await globby(["./src/docs/**/*.md"], { dot: true });
+  mdFiles.forEach(transformMarkdown);
+
+  console.log("Transforming html files...");
+  const htmlFiles = await globby(["./src/docs/**/*.html"], { dot: true });
+  htmlFiles.forEach(transformHtml);
+
+  console.log("Transforming all other files...");
+  const otherFiles = await globby(["src/docs/**", "!**/*.md", "!**/*.html"], { dot: true });
+  otherFiles.forEach((file) => {
     verifyAndCopy(file);
   });
-  if (fileChanged) {
-    if (verify) {
+  if (filesWereChanged) {
+    if (verifyOnly) {
       console.log(
-        "Changes detected in files in `docs`. Please run `yarn docs:build` after making changes to 'src/docs' to update the `docs` folder."
+        "Changes detected in files in `src/docs`. Please run `yarn docs:build` after making changes to 'src/docs' to update the `docs` folder."
```

Comment:
This should be `docs` itself. The most common scenario is someone just edits `docs`. 

How about 

### Generated `docs` differ from their source `src/docs`. Please run `yarn docs:build` after making changes to `src/docs` to update the `docs` folder.

Reply:
Makes sense. (doh! I was interpreting the log message as just informing the user about what src/docs had changed. Which of course user would know, having made the changes!)

Reply:
Actually, looking at the code again...
The way the code is now, the message is logged to the console if the _transformed_ `src/doc` file is different from the `/doc` file.  IOW, `/doc` is being compared to the _already transformed_ contents (not the original /src/doc file before it was transformed).
That's why I thought the console message was just informative.

So is the purpose of the `verifyAndCopy` method simply to verify that the file in `/docs` has not been changed?

Reply:
`verifyAndCopy` does 2 things.

- Check if the file has been changed & fail if `--verify` is passed.
- Overwrite file if there are changes and `--verify` is not passed.

Reply:
```mermaid
graph
  A[docs:build] --> B[Transform `src/docs`]
  B --> C[Compare with `docs`]
  C --> D{Has changes?}
  D -->|No| Done
  D -->|Yes| E{--verify passed?}
  E -->|No| Overwrite
  Overwrite --> Done
  E-->|Yes| Fail
```

Reply:
thanks for the flowchart. 😄 

Seems that there are 2 different things than need to be checked:
1. Has any file in `/docs` been changed?
    - If so, then warn the user and stop. _No need to continue at all_
2. Is a transformed file (contents) the same as the `/docs` file?
    - if it is the same, then the `/docs` file is up to date and the transformed file does not need to be copied to /docs
    - else 
       - copy the transformed file to `/docs`

Right now the code is only doing #2 

```mermaid
   flowchart
   style start stroke:green,stroke-width:4px,color:green,fill:white
  style fail stroke:red,color:red,fill:white
  style success stroke:green,color:greenfill:white

     start[Assume we only have 1 documentation file: mermaid.md]
      docs:build --> checkDocs{Has any file in `/docs` been changed?}
      checkDocs --> |Yes| consoleChangeMsg["Warn user than /docs files were changed\n(list them)"]
     consoleChangeMsg --> fail[Fail] 
     checkDocs --> |No| transformedSameAsDocs{"Is the transformed mermaid.md\nthe same as /docs/mermaid.md ?"}
transformedSameAsDocs -->|No| verifyOption{Was the --verify option passed in?}
verifyOption --> |Yes\nThis is confusing to me. \nWhy return a failure? What is it _verifying_?| fail 
verifyOption --> |No| copy["copy the transformed mermaid.md\nto /docs/mermaid.md\n (overwrite)"] --> success
transformedSameAsDocs -->|Yes| docsCurrent["/docs/mermaid.md is up to date. nothing to do"]
docsCurrent --> success(("success"))

```
**Is the flow chart above right**

Also:
If the `--verify` flag is passed in and  the `/docs` files are up to date, then why return a fail code?  if all it has done is to verify that everything is ok, then why not return a success?

Also, I think the option would be better renamed to `--verifyOnly` (assuming I have the intention right. I might not, hence my confusion.)

Reply:
> If the --verify flag is passed in and the /docs files are up to date, then why return a fail code?

It doesn't. 

> This is confusing to me.
Why return a failure? What is it _verifying_?

It's verifying if `transformed('src/docs') == 'docs'`

The current simple flow is:
- User changes `src/docs`, their pre-commit hook automatically commits changes to `docs`.
- They create a PR with changes in both `src/docs` and `docs`
- CI verifies `transformed('src/docs') == 'docs'`


A more complex flow which would enable users to easily edit via GitHub UI would be:
- User edits `src/docs`
- Creates PR with just that edit
- CI builds `docs` and pushes to the PR if there are no changes to files in `docs` since the last commit by our CI.
- CI fails if there are changes to `docs`. 


But `CI builds docs and pushes to the PR if there are no changes to files in docs since the last commit by our CI.` is a bit complex. There are some cases we have to consider:
- If the user checks out the branch locally, edits `src/docs`, then the precommit hook will build and commit `docs` too. So that check will fail. (Possible fix, edit precommit hook & enforce it to fail if there are changes to `docs`. But we know precommit hooks aren't that effective)
- Not everyone will be very familiar with git to revert commits, so we'll have to check by file content since last commit by CI and not just commits. 


I'm pretty sure there'll be some simpler way, so let's brainstorm?

Reply:
Aha!  This was the key to my confusion:

> It's verifying if transformed('src/docs') == 'docs'

This only tells us if they are different; it doesn't tell us if `docs` was changed or if `src/docs` was changed.

> The most common scenario is someone just edits docs.

I was trying to also check for this in the flow chart I did. (specifically in "Has any file in `/docs` been changed?")

I'll make another pass at changing the code in this PR so that it matches _"The current simple flow is:..."_ above.


How about if we talk about/brainstorm the larger CI/development flow in either a new issue  or a new discussion?
(And as a part of that we should look at the CI/development workflows that other projects use to generate documentation.)

Reply:
> I'll make another pass at changing the code in this PR so that it matches "The current simple flow is:..." above.

Sure, I actually didn't notice you had changed the verification flag's working. Let's stick with the previous flow and discuss the complex one in a new issue.